### PR TITLE
deps: Switch `envoy.dependency.pip_check` -> `dependatool`

### LINKED ISSUE #26163: Add dependabot checks for Docker/go
We currently check that the dependabot config matches `requirements.txt` in the repo

we should do simiar for `Dockerfile*` and `go.mod`

### PR DESCRIPTION
**Pull request by phlax**
the `pip_check` tool has been renamed to `dependatool` and includes docker and go checks and related fixes

Fix #26163 

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .github/dependabot.yml
Comment (RyanTheOptimist) [inline .github/dependabot.yml:1]: I don't think i understand the changes in this file. What's the story with them?

Comment (phlax) [reply .github/dependabot.yml:1]: dependabot.yml controls which dependency files (requirements.txt, go.mod, Dockerfile) dependabot looks at to update 

when we originally set up dependabot we had a lot of python requirements.txt files (thankfully reduced now) so we added a tool to make sure that dependabot.yml <> files matched

now we have a lot of Dockerfiles and increasingly go.mod files also

the `pip_check` tool has been renamed to `dependabot` and has added functionality for these additional file types

running the tool in CI it expects the dependabot.yml to be correct - hence the changes

**Comment by phlax** [comment]
```console
DependatoolChecker SUCCESS [gomod] ✔ go.mod: /examples/golang/simple
DependatoolChecker SUCCESS [gomod] ✔ go.mod: /examples/grpc-bridge/server
DependatoolChecker SUCCESS [gomod] ✔ go.mod: /examples/load-reporting-service
DependatoolChecker ERROR [gomod] No go.mod found for specified dependabot config: /contrib/golang
DependatoolChecker ERROR [gomod] Missing dependabot config for go.mod in dir: /
DependatoolChecker SUCCESS [pip] ✔ requirements.txt: /.github/actions/pr_notifier
DependatoolChecker SUCCESS [pip] ✔ requirements.txt: /examples/cache
DependatoolChecker SUCCESS [pip] ✔ requirements.txt: /examples/grpc-bridge/client
DependatoolChecker SUCCESS [pip] ✔ requirements.txt: /examples/shared/python
DependatoolChecker SUCCESS [pip] ✔ requirements.txt: /examples/shared/python/postgres
DependatoolChecker SUCCESS [pip] ✔ requirements.txt: /mobile/docs
DependatoolChecker SUCCESS [pip] ✔ requirements.txt: /tools/base
DependatoolChecker SUCCESS [pip] ✔ requirements.txt: /tools/code_format
DependatoolChecker ERROR ERRORS Summary [gomod]:
--------------------------------------------------------------------------------
No go.mod found for specified dependabot config: /contrib/golang
Missing dependabot config for go.mod in dir: /
```

**Comment by phlax** [comment]
cc @doujiang24 i thought this would break on #26606 and raised https://github.com/envoyproxy/pytooling/pull/703 to address but it turns out the file disco tool already filters empty files so it just works

**Review by RyanTheOptimist — APPROVED** [review]

**Pull request merged**